op-e2e/interop/interop_benchmark_test.go

```diff
@@ -0,0 +1,214 @@
+package interop
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/interopgen"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/rand"
+)
+
+// setupAndRun is a helper function that sets up a SuperSystem
+// which contains two L2 Chains, and two users on each chain.
+func setup(t testing.TB, config SuperSystemConfig) SuperSystem {
+	recipe := interopgen.InteropDevRecipe{
+		L1ChainID:        900100,
+		L2s:              []interopgen.InteropDevL2Recipe{{ChainID: 900200}, {ChainID: 900201}},
+		GenesisTimestamp: uint64(time.Now().Unix() + 3), // start chain 3 seconds from now
+	}
+	worldResources := WorldResourcePaths{
+		FoundryArtifacts: "../../packages/contracts-bedrock/forge-artifacts",
+		SourceMap:        "../../packages/contracts-bedrock",
+	}
+
+	// create a super system from the recipe
+	// and get the L2 IDs for use in the test
+	s2 := NewSuperSystem(t, &recipe, worldResources, config)
+
+	// create two users on all L2 chains
+	s2.AddUser("Alice")
+	s2.AddUser("Bob")
+	s2.AddUser("Charlie")
+	s2.AddUser("Dennis")
+	s2.AddUser("Eve")
+	s2.AddUser("Frank")
+
+	return s2
+}
+
+func BenchmarkCheckMessages(b *testing.B) {
+	// Skip this benchmark
+	b.Skip("Run benchmarks on demand only")
+
+	// Set up the SuperSystem once before the benchmark loop
+	s2 := setup(b, SuperSystemConfig{})
+
+	// Set up chains with Emitter Contracts
+	chains := s2.L2IDs()
+	chainA := chains[0]
+	chainB := chains[1]
+	clientA := s2.L2GethClient(chainA, "sequencer")
+	clientB := s2.L2GethClient(chainB, "sequencer")
+	// Deploy emitter to chain A
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	EmitterA := s2.DeployEmitterContract(ctx, chainA, "Alice")
+
+	// Deploy emitter to chain B
+	ctx, cancel = context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	EmitterB := s2.DeployEmitterContract(ctx, chainB, "Alice")
+
+	// Set up chains with lots of initiating messages
+	numEmits := 100
+	// emit logs on both chains in parallel
+	var emitParallel sync.WaitGroup
+	emitOn := func(chainID string, address string) {
+		for i := 0; i < numEmits; i++ {
+			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+			payload := fmt.Sprintf("%s-%d", address, i)
+			s2.EmitData(ctx, chainID, "sequencer", address, payload)
```

Comment:
I think this implies that there is at most 6 logs per block, since per user it confirms the EmitData?
It would be nice to program a benchmark with more logs per block.

The devstack + txplan code has a bunch of utils to prepare chained txs, confirm them together in parallel, and bundle multiple log-events into the same tx also.

This can be split out in a follow-up PR though.

Reply:
Yes, the wait-for-mined meant the callers were limited to one tx per block, which is why I added 6 users to each chain. Unfortunately it also makes the test setup really slow since there are implicitly `numEmits` blocks to mine.

Good to hear the devstack will improve this.